crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -553,7 +551,8 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientState {
                                             .unwrap()
                                             .handle_input(input.clone());
 
-                                        wait_time = Duration::from_millis(1000 / rate as u64);
+                                        let repeat_delay = Duration::from_secs(1) / rate;
+                                        Timer::after(repeat_delay).await;
```

Comment:
I think we'll need to drop the `state` before the `.await`, otherwise the borrow is still alive.